### PR TITLE
Add: Sync  Plugin Updated event.

### PR DESCRIPTION
**Pull request by enejb**
This PR adds test for plugin updates and plugin update failures.

#### Changes proposed in this Pull Request:

- Fix API for plugin updates so that it returns an error as expected. 
- Adds 
  - syncing of plugin updates on success fully plugin update and failure.
  - updates the ....wait for it....
 

#### Testing instructions:
* Do the tests pass? 
* make it so that plugin updates fail ( use a plugin like something below) 
```
<?php
/*
 * Plugin Name: Break Plugin Updates
*/

add_filter( 'upgrader_package_options', 'eb_break_plugin_updated' );
function eb_break_plugin_updated( $options ) {
        $options['package'] = 'http://example.com';
        return $options;
}
```
- try updating a plugin via the api (in calyspo) Do you notice an error? 
- try updating the plugin via the wp-admin interface, plugin.php and core-updates.php UI.
- try updating the plugin via the wp cli interface. `wp plugin update jetpack`
Do you get the expected failure event on .com? 

- Try updating the plugin do you get the event as expected on .com that tells you that a plugin updated? 

- Do you still get a plugin install event as expected?

### CODE REVIEW

**Comment by lezama** [comment]
nice work @enejb, @zinigor, @dereksmart could we ship this one before the freeze?

**Comment by enejb** [comment]
Thanks for review @dereksmart. I will test all the steps some more but I think this is good to go I fixed the error you found above!

**Comment by dereksmart** [comment]
Changes look good! Works as expected now

File: sync/class.jetpack-sync-module-plugins.php [hunk 1]
```diff
@@ -30,68 +31,129 @@ public function init_before_send() {
...
+			 * @since 5.8.0
+			 *
+			 * @module sync
 			 *
-			 * @param string $plugin_path Path of plugin installed
-			 * @param mixed $plugin_info Array of info describing plugin installed
+			 * @param array() $plugin, Plugin Data
 			 */
-			do_action( 'jetpack_installed_plugin', $plugin_path, $plugin_info );
+			do_action( 'jetpack_plugin_installed',  array_map( array( $this, 'get_plugin_info' ), $plugins ) );
```
Comment (zinigor) [inline sync/class.jetpack-sync-module-plugins.php]: Hate to nitpick that late in the process, but do you think we need to keep the old action as a deprecated action?

**Pull request merged**